prime-dotnet-webapi/Services/ReportingService.cs

```diff
@@ -0,0 +1,117 @@
+using Microsoft.EntityFrameworkCore;
+using Microsoft.Extensions.Logging;
+using System;
+using System.Linq;
+using System.Threading.Tasks;
+
+using Prime.Models;
+using Prime.HttpClients;
+
+namespace Prime.Services
+{
+    public class ReportingService : BaseService, IReportingService
+    {
+        private const int CalculationPeriodInDays = 14;
+
+        private readonly ICollegeLicenceClient _collegeLicenceClient;
+
+        public ReportingService(
+            ApiDbContext context,
+            ILogger<ReportingService> logger,
+            ICollegeLicenceClient collegeLicenceClient)
+            : base(context, logger)
+        {
+            _collegeLicenceClient = collegeLicenceClient;
+        }
+
+        public async Task<string> PopulatePractitionerTableAsync(DateTime? startDate = null, DateTime? endDate = null)
+        {
+            var result = "";
+            try
+            {
+                if (startDate == null)
+                {
+                    startDate = DateTime.Now.AddDays(-CalculationPeriodInDays);
+                }
+
+                if (endDate == null)
+                {
+                    endDate = DateTime.Now;
+                }
+
+                // get all approved enrollee
+                var enrolleeCerts = _context.Enrollees
+                    .Where(e => e.GPID != null && e.Certifications.Any())
+                    .Select(e => new
+                    {
+                        e.Certifications.FirstOrDefault().LicenseNumber,
+                        e.Certifications.FirstOrDefault().Prefix
```

Comment:
`Certification.Prefix` may be `null`

Reply:
Interesting observation. I did a quick query in Prod PharmaNet Translation log and there is no record where College Prefix is null and Practitioner ID is not null. 
Updated the query to return the ones Prefix is not null.

Reply:
To clarify, should it be like:  https://github.com/bcgov/moh-prime/blob/c12e19a5d4480bce8660fbc7a1f6318fffab1eb6/prime-dotnet-webapi/Services/EnrolleeService.cs#L1072C70-L1072C70